client/src/components/__tests__/FormDialog.test.js

```diff
@@ -37,15 +37,13 @@ describe('<FormDialog />', () => {
 
   test('should contain cancel and submit button', () => {
     const buttons = wrapper.find('button');
-    expect(
-      buttons
-        .at(0)
-        .text()
-        .toLowerCase(),
-    ).toContain('cancel');
-
-    expect(buttons.at(1).contains(props.title)).toBe(true);
-    expect(buttons.at(1).props().type).toBe('submit');
+
+    const cancelBtn = buttons.at(0);
```

Comment:
Let me look into it!

Reply:
I'm facing an issue here;

I added data-test attribute to the button:

`js <Button onClick={closeModal} color="primary" data-test="cancel">Cancel<Button>
`
and the if I do: 

```js
    const cancelBtn = wrapper.find('[data-test="cancel"]');
    expect(cancelBtn.text().toLowerCase()).toContain('cancel');

```
The test fails with: 

>  Method “text” is meant to be run on 1 node. 5 found instead.

However, 

If I do the following, it works: 

```js
    const cancelBtn = wrapper.find('[data-test="cancel"]').at(0);
    expect(cancelBtn.text().toLowerCase()).toContain('cancel');

```

Is this an expected behavior?

Reply:
NVM, I figured out, I also have to mention the component name I guess:

```js
const cancelBtn = wrapper.find('Button[data-test="cancel"]');
```

Reply:
> Method “text” is meant to be run on 1 node. 5 found instead.

How is it showing that you have 5 cancel buttons?

Reply:
I'm guessing because it is giving the same "data-test" to all children elements too.

Reply:
But form-dialog has only one `data-test="cancel"`

Reply:
The `<Button/>` component is from material-ui and it might have multiple children. And those children themselves might have children. I'm doing the deep rendering instead of shallow so all of them get the data-test.

Reply:
That's just my guess.

Reply:
where have you added the data-test attribute ?

Reply:
Here: 

https://github.com/pesto-students/rattle-battle/blob/64d93ef9e3cfd0bac21378719532f4e92d2e4b9f/client/src/components/FormDialog.js#L17

Reply:
Yes here you have to use `Button[data-test="cancel"]`